### ISSUE TITLE
NQ Test CTX+ data

### ISSUE THREAD

**Issue by krishanudb** [facebookresearch/DPR#97]
Hello,
I am working on replicating your results and doing some more analyses on it.
Will it be possible for you to provide the positive contexts / passages for the NQ test set (from the passages indexed). It would be beneficial for calculating the recall and precision score easily since the gold contexts / passages given in the nq-test are a bit different from the passages indexed (psgs_w100 file).

P.S. I tried to code up a solr/lucene index using the psgs_w100.tsv file and search and find the most relevant documents to the gold_contexts. I was planning to use these retrieved passages as the CTX+ passages. However, there seems to be very low matches. I might be missing something.

Hence, I would be obliged to you if you can provide the code for creating the ctx+ passages for the question using the gold positive passages, or you can share the preprocessed data.

Thanks,
Krishanu.

**Comment by vlad-karpukhin** [comment]
Hi, 

We didn't use bm25 passages for all test sets. 
A test set for the retriever inference time logic is a simply a list of question-> [answers]  items In ODQA setting.
We just retrieve top-k passages from generated Wikipedia embeddings and apply our reader model on top of them.
So, I didn't quite understand why you need positive contexts / passages for the NQ test set.

The gold to psg-w100 file mapping process we used is highly fragmented, hardcoded and suitable for our filesystem/infrastructure so we can't provide anything useful for public for this task, sorry.

I do believe however, that a better hard negative and positive passages can be mined from our dense model& index itself instead of from bm25 index and then fine-tune or train our retriever from scratch.

**Comment by krishanudb** [comment]
Hi,

Thanks for replying.

I wanted to evaluate the retriever on unseen data, using metrics like precision, recall, etc. My research is partially on how to deal with noise in retriever model outputs and so I thought getting these numbers on the DPR model would be useful.

Thanks,
Krishanu.
